runtime/litentry/src/migration/balances_transaction_payment.rs

```diff
@@ -0,0 +1,383 @@
+pub const DECIMAL_CONVERTOR: u128 = 1_000_000u128;
+
+#[cfg(feature = "try-runtime")]
+use parity_scale_codec::Encode;
+#[cfg(feature = "try-runtime")]
+use sp_std::collections::btree_map::BTreeMap;
+use storage::migration::get_storage_value;
+
+// Replace Frame System Storage for Decimal Change from 12 to 18
+// Replace Balances Storage for Decimal Change from 12 to 18
+pub struct ReplaceBalancesRelatedStorage<T>(PhantomData<T>);
+impl<T> ReplaceBalancesRelatedStorage<T>
+where
+	T: frame_system::Config<AccountData = AccountData<u128>>
+		+ pallet_balances::Config<Balance = u128>,
+{
+	pub fn replace_frame_system_account_storage() -> frame_support::weights::Weight {
+		// The storage of Account for pallet balances is in frame_system pallet
+		log::info!(
+			target: "ReplaceBalancesRelatedStorage",
+			"running migration to Frame System Account"
+		);
+		let pallet_prefix: &[u8] = b"System";
+		let storage_item_prefix: &[u8] = b"Account";
+		let mut weight: Weight = frame_support::weights::Weight::zero();
+
+		for (account, mut account_info) in storage_key_iter::<
+			T::AccountId,
+			AccountInfo<T::Index, T::AccountData>,
+			Blake2_128Concat,
+		>(pallet_prefix, storage_item_prefix)
+		.drain()
+		{
+			account_info.data.free = account_info.data.free.saturating_mul(DECIMAL_CONVERTOR);
+			account_info.data.reserved =
+				account_info.data.reserved.saturating_mul(DECIMAL_CONVERTOR);
+			account_info.data.frozen = account_info.data.reserved.saturating_mul(DECIMAL_CONVERTOR);
+
+			<Account<T>>::insert(&account, account_info);
+			weight += T::DbWeight::get().reads_writes(1, 1);
+		}
+		weight
+	}
+	pub fn repalce_balances_total_issuance_storage() -> frame_support::weights::Weight {
+		log::info!(
+			target: "ReplaceBalancesRelatedStorage",
+			"running migration to Balances TotalIssuance"
+		);
+		let pallet_prefix: &[u8] = b"Balances";
+		let storage_item_prefix: &[u8] = b"TotalIssuance";
+		let stored_data = get_storage_value::<u128>(pallet_prefix, storage_item_prefix, b"")
+			.expect("Storage query fails: Balances TotalIssuance");
+		<TotalIssuance<T>>::put(stored_data.saturating_mul(DECIMAL_CONVERTOR));
+		let weight = T::DbWeight::get();
+		frame_support::weights::Weight::from_parts(0, weight.write + weight.read)
+	}
+	pub fn repalce_balances_inactive_issuance_storage() -> frame_support::weights::Weight {
+		log::info!(
+			target: "ReplaceBalancesRelatedStorage",
+			"running migration to Balances InactiveIssuance"
+		);
+		let pallet_prefix: &[u8] = b"Balances";
+		let storage_item_prefix: &[u8] = b"InactiveIssuance";
+		let stored_data = get_storage_value::<u128>(pallet_prefix, storage_item_prefix, b"")
+			.expect("Storage query fails: Balances InactiveIssuance");
+		<InactiveIssuance<T>>::put(stored_data.saturating_mul(DECIMAL_CONVERTOR));
+		let weight = T::DbWeight::get();
+		frame_support::weights::Weight::from_parts(0, weight.write + weight.read)
+	}
+	pub fn check_balances_account_storage() -> frame_support::weights::Weight {
```

Comment:
I'm not sure if we even want this fn (also other fields check).

We expect it to be empty, but `assert!` will panic if it's non-empty, won't it? I don't know how it can be non-empty, just somehow.

Reply:
No way it can be non-empty.
I have removed the runtime upgrade part but keep the pre/post code.